### PR TITLE
fixed some mortar tests

### LINKED ISSUE #6488: correct some mortar tests
### Description of the enhancement or error report
some mortar tests are not running correctly. 

### Rationale for the enhancement or information for reproducing the error
There are two issues for some mortar tests.

1. For a well-posed interface problem, two interfacial conditions are needed, the jump in the field ```[[u]] = i``` and the jump in the flux ```[[\grad u \dot n]] = j```. For the weak form, integration by part will give a jump term ```\int [[\grad u \dot n]] * <test_function>``` on the interface. With the given value ```j```, this term becomes ```\int j * <test_function>``` and goes to the RHS. No bc kernel is needed on the interface.

1. Mortar method uses ```Lagrange multiplier```. For this mixed formulation the discrete spaces for the discretization of the primal variable and the multipliers have to fulfill the inf-sup condition in order the resulting numerical scheme to be stable. Empirically the multiplier should use lower order shape function than primal variable's one. For one mortar test in which both primal variable and the multiplier use second lagrange shape function, the solve does not converge. If I change to the multiplier shape function to first order, the solve converges pretty well.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)
only test files are modified

### PR DESCRIPTION
**Pull request by jiangwen84**
Fixing some mortar tests. 
1. Remove unnecessary DiffusionFluxBC
2. Reduce the shape function order for Lagrange multiplier variable where needed. Now all the tests converge better. 
3. Regolded some tests.    

closes  #6488

### CODE REVIEW

File: test/tests/mortar/2d/tests [hunk 1]
```diff
@@ -23,7 +23,7 @@
     exodiff = 'non-conforming_out.e'
     max_parallel = 1
     max_threads = 1
-    deleted = true # Diffs 25% of the time in a significant way #1482
+    # deleted = true # Diffs 25% of the time in a significant way #1482
```
Comment (andrsd) [inline test/tests/mortar/2d/tests]: I'd suggest to remove this instead of commenting out.

Comment (jiangwen84) [inline test/tests/mortar/2d/tests]: Yes, will do.

**Comment by andrsd** [comment]
You want to do `git rebase --interactive HEAD~2` and mark the second patch as fixup. That will give you a branch with one clean patch. Then force push your branch...

**Comment by andrsd** [comment]
:+1:

**Pull request merged**